adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/authentication/KeycloakAuthenticationEntryPoint.java

```diff
@@ -105,7 +106,13 @@ protected void commenceLoginRedirect(HttpServletRequest request, HttpServletResp
             // stored in a session. We'll store it in a cookie instead.
             response.addCookie(KeycloakCookieBasedRedirect.createCookieFromRedirectUrl(request.getRequestURI()));
         }
-        String contextAwareLoginUri = request.getContextPath() + loginUri;
+
+        String queryParameters = "";
+        if (!StringUtils.isEmpty(request.getQueryString())) {
+            queryParameters = "?" + request.getQueryString();
```

Comment:
I am not sure we want to add all parameters. Maybe we should use only those that can be consumed by Keycloak. WDYT?

Reply:
Currently the https://github.com/keycloak/keycloak/blob/cd342ad5714f15db1cc8b0cd55b788e6543c6dc8/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java will check which query parameters are moved up to top url (not redirect uri), e.g. prompt=login. All other parameters are passed as is.
For us (my company) we define the language the application should use by setting the query param "lang=en" on the application url. If we don't strip this parameter we can check this in our Keycloak authenticators too.
In addition now it's the same behavior as in the JBoss adapter.

Reply:
You are right @dweber019, this makes sense. Thank you very much for further explanation.